### PR TITLE
Drag & Drop: Disable Drag & Drop into Empty Placeholders

### LINKED ISSUE #6035: Drag & dropping blocks shows 'Drop files to upload' zones
When drag and dropping blocks, the 'Drop files to upload' region appears. In the context of it being a block you are moving, this makes no sense in appearing. Refer to screenshot for reference.

screen shot 2018-04-06 at 12 52 43

### PR DESCRIPTION
**Pull request by youknowriad**
closes #6035 

Allowing Drag & Drop into the empty placeholders is a bit confusing because their styling is different and their label is misleading.

In this PR, I'm updating the style of the default block appender dropzone to match the regular block's dropzone (just a single line).

**Testing instructions**

 - try drag and dropping a block
 - While dragging, the dropzone with the "Drop file to upload" label is not shown.

### CODE REVIEW

File: editor/components/block-drop-zone/index.js [hunk 1]
```diff
@@ -84,51 +79,55 @@ class BlockDropZone extends Component {
...
-					dispatch( updateBlockAttributes( ...args ) );
-				},
-				moveBlockToPosition( uid, fromRootUID, index ) {
-					const { rootUID, layout } = ownProps;
-					dispatch( moveBlockToPosition( uid, fromRootUID, rootUID, layout, index ) );
-				},
-			};
-		},
-	),
+	withDispatch( ( dispatch, ownProps ) => {
```
Comment (youknowriad) [inline editor/components/block-drop-zone/index.js]: Since I touched the file, I refactored it to use `withDispatch` instead of `connect`

**Comment by paulwilde** [comment]
Just tested and it's fixed the issue.

**Comment by SuperGeniusZeb** [comment]
Wouldn't this make it impossible to drag a block into an empty column?

**Comment by youknowriad** [comment]
Good catch @SuperGeniusZeb I'll try to find an alternative fix

**Comment by youknowriad** [comment]
Updated, it should be better now.

File: editor/components/block-drop-zone/style.scss [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+		}
+
+		&.is-close-to-top,
+		&.is-appender.is-close-to-top,
+		&.is-appender.is-close-to-bottom {
+			background: none;
+			border-top: 3px solid $blue-medium-500;
+			border-bottom: none;
+		}
+	}
```
Comment (noisysocks) [inline editor/components/block-drop-zone/style.scss]: nit: This file is one tab too indented.

**Review by noisysocks — APPROVED** [review]
Nit aside, this looks good to me.

I tested:

- Dragging blocks around
- Dragging an image file into the post
- Dragging a block into a Columns block

**Comment by noisysocks** [comment]
I maybe spoke too soon. I've noticed this glitch:

1. Insert a paragraph block
2. Insert a Columns block
3. Insert an image into the right column
4. Drag the paragraph block to the bottom of the post
5. Note that you can see some white 'Drop file to upload' text appear over the image:  
  screen shot 2018-04-09 at 12 11 11

**Comment by youknowriad** [comment]
Good catch @noisysocks It should behave better now.

**Pull request merged**